go/vt/topo/shard.go

```diff
@@ -121,6 +121,10 @@ func IsShardUsingRangeBasedSharding(shard string) bool {
 // ValidateShardName takes a shard name and sanitizes it, and also returns
 // the KeyRange.
 func ValidateShardName(shard string) (string, *topodatapb.KeyRange, error) {
+	if strings.Contains(shard, "/") {
+		return "", nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid shardId, may not contain '-': %v", shard)
```

Comment:
I see that the error code here is the correct one - `INVALID_ARGUMENT`. Why does the client display `Unknown`?

Reply:
that's the vtadmin-web level of reporting, it's INVALID_ARGUMENT all the way through to `vtadmin-api`

Reply:
https://github.com/vitessio/vitess/blob/main/go/vt/vtadmin/http/response.go#L41-L61

Reply:
Do we convert all RPC errors to Unknown errors? It just seems misleading to be claiming that it is an `unknown rpc error` when it is not.
Or is this deliberate obfuscation?

Reply:
it depends (NOT_FOUND gets converted to 404, for example), but that's something we can address separately if you feel it's an issue